src/middleware/fileupload.js

```diff
@@ -1,56 +1,60 @@
+                                console.log(line_array[count])
+                                var fetched_data = {
+                                    id: line_array[count][0],
+                                    user_id: line_array[count][2],
+                                    timing: line_array[count][6]
+                                }
+                                var flag = 0;
+                                _.forEach(filtered_data, function(value) {
 
-                        function readFile(callback) {
-                            lineReader.on('line', function(line) {
-                                line_array.push(line.split("\t"))
-                                if (line_array[count][0] && count >= 1) {
-                                    var fetched_data = {
-                                        No: line_array[count][0],
-                                        TMNo: line_array[count][1],
-                                        EnNo: line_array[count][2],
-                                        Name: line_array[count][3],
-                                        GMNo: line_array[count][4],
-                                        Mode: line_array[count][5],
-                                        "IN/OUT": line_array[count][6],
-                                        Antipass: line_array[count][7],
-                                        DateTime: line_array[count][8]
+                                    if (value.id == fetched_data.id || (fetched_data.user_id).toString().trim() == "0") {
+                                        flag = 1;
                                     }
-                                    filtered_data.push(fetched_data);
+                                })
+
+                                if (flag == 0) {
+                                    filtered_data.push(fetched_data); // array of non repeative data from file
                                 }
-                                count++;
-                            });
-                            lineReader.on('close', function() {
-                                callback(filtered_data);
-                            })
-                        }
-                        readFile(function(response) {
-                            model.details.bulkCreate(response).then(function() {
-                                fs.unlink(newpath);
-                                res.json('File uploaded and data moved!');
-                            })
+                            }
+                            count++;
+                        });
+                        lineReader.on('close', function() {
+                            callback(filtered_data);
                         })
                     }
-                });
-            });
+                    readFile(function(response) {
+                        db.details.bulkCreate(response).then((data) => { // insert data into database
```

Comment:
what does this do? write in comments

Reply:
sir,this function accepts filtered data array as callback and insert it to db (bulk create) at once .